includes/class-amp-story-post-type.php

```diff
@@ -1500,4 +1512,29 @@ public static function change_embed_iframe_attributes( $output, $post ) {
 			$output
 		);
 	}
+
+	/**
+	 * Adds a new max image size to the images sizes available, and removes 'full'.
+	 *
+	 * In the AMP story editor, when selecting Background Media,
+	 * it will use this custom image size.
+	 * This will also be available for image blocks.
+	 * But the new size will only be available if the original image was at least as big as it is.
+	 *
+	 * @param array $image_sizes {
+	 *     An associative array of image sizes.
+	 *
+	 *     @type string $slug Image size slug, like 'medium'.
+	 *     @type string $name Image size name, like 'Medium'.
+	 * }
+	 * @return array $image_sizes The filtered image sizes.
+	 */
+	public static function add_new_max_image_size( $image_sizes ) {
+		if ( self::POST_TYPE_SLUG === get_post_type() ) {
+			unset( $image_sizes['full'] );
```

Comment:
Why remove the `full` size since the story background image size is being explicitly looked for?

Reply:
Yeah, the `unset()` statement has no effect on the background image, as it's not using the `full` size.

This filter applies to the story background image *and* the Image block.

That `unset()` statement prevents the user from selecting the `'full'` size image in the Image block:

<img width="1225" alt="larg-here" src="https://user-images.githubusercontent.com/4063887/59088886-658eab00-88ce-11e9-86d6-9fe8007e072c.png">

For the Image block, this PR doesn't change the default size that's used. That's still the `'large'` size, which is [usually 1024 x 1024px max](https://codex.wordpress.org/Post_Thumbnails#Thumbnail_Sizes)). Though I could look at changing that.

If this PR didn't remove the `full` size, it wouldn't help much for the Image block. 

It would add an intermediate size that users could choose, which is slightly bigger than the `large` size. But they could still choose the full image.